packages/sdk/src/infrastructure/solana-utils/payment/paystream/PaystreamStartService.ts

```diff
@@ -0,0 +1,299 @@
+
+  private startedAt?: number;
+
+  private endedAt?: number;
+
+  private paymentAccount?: Keypair;
+
+  private maxTime = 0;
+
+  protected async init(props: CreatePaystreamProps) {
+    await super.init(props);
+
+    this.startedAt = this.dateToTimeStamp(new Date());
+    this.maxTime = props.maxTime;
+    this.endedAt = this.startedAt + this.maxTime;
+    this.paymentAccount = new Keypair();
+  }
+
+  protected async executeTransaction(
+    requestOrPaymentId: string,
+    {
+      request,
+      symbol,
+      fixedCurrencyRateToken,
+      swapRouteToken,
+      canSwapTokens,
+    }: ExecuteTransactionPayload<CreatePaymentRequest>
+  ): Promise<SignedTxAndToken> {
+    if (this.connection && this.wallet) {
+      if (swapRouteToken && canSwapTokens) {
+        return this.getAndSignCreateSwapPaymentTx({
+          wallet: this.wallet,
+          request,
+          symbol,
+          paymentRequestId: requestOrPaymentId,
+          swapRoute: swapRouteToken,
+          fixedCurrencyRateToken,
+        });
+      }
+      return this.getAndSignCreatePaymentTx({
+        wallet: this.wallet,
+        request,
+        symbol,
+        paymentRequestId: requestOrPaymentId,
+        fixedCurrencyRateToken,
+      });
+    }
+    throw new Error('Connection or wallet is not defined');
+  }
+
+  protected getTransactionParams({
+    amount,
+    anchorProvider,
+    recipientPK,
+    interval,
+  }: CreatePaystreamProps): CreatePaymentRequest {
+    return {
+      amount: String(amount),
+      sender: anchorProvider.provider.publicKey as PublicKey,
+      recipient: new PublicKey(recipientPK),
+      mintAddress: this.mintAddress,
+      paymentAccount: this.paymentAccount?.publicKey as PublicKey,
+      startAt: String(this.startedAt),
+      endAt: String(this.endedAt),
+      interval,
+    };
+  }
+
+  protected getTransactionPayload(
+    {
+      paymentRequestId,
+      amount,
+      anchorProvider,
+      recipientPK,
+      symbol,
+      customerDetails,
+      productDetails,
+      interval,
+      rateToken,
+    }: CreatePaystreamProps,
+    signedTx: string,
+    token: string,
+    swapSignedTx?: string
+  ): ApproveStreamTransactionPayload {
+    const details = isEmptyObject(productDetails) ? {} : { productDetails };
+    return {
+      signedTransaction: signedTx,
+      signedSwapTransaction: swapSignedTx,
+      streamToken: token,
+      paymentRequestId,
+      amount: fromBigintToStringForSerialization(BigInt(amount)),
+      sender: (anchorProvider.provider.publicKey as PublicKey).toBase58(),
+      recipient: recipientPK,
+      currency: symbol,
+      customerDetails,
+      paymentAccount: (this.paymentAccount as Keypair).publicKey.toBase58(),
+      startedAt: this.startedAt as number,
+      endedAt: this.endedAt as number,
+      interval,
+      rateToken,
+      ...details,
+      cluster: this.cluster as Cluster,
+    };
+  }
+
+  protected onSuccess(
+    response: CreatePaystreamAPIResponse,
```

Comment:
please make sure this response is correct and we get it from return value of common package